### PR TITLE
MINOR: Improve release script

### PR DESCRIPTION
**Pull request by jlprat**
These are the fixes for the problems I encountered while creating the RC for the 3.8.0 version.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jlprat** [comment]
@soarez these are the changes we discussed already.

File: release/README.md [hunk 1]
```diff
@@ -52,3 +52,9 @@ Then start the release script:
 python release.py
 ```
 
+Should you encounter some problem during the script re-running doesn't work look at the following steps:
```
Comment (soarez) [inline release/README.md]:
```suggestion
Should you encounter some problem, where re-running the script doesn't work, look at the following steps:
```

**Review by soarez — COMMENTED** [review]
Thanks for the improvements @jlprat . Just a minor comment on wording, but otherwise looks good.

**Comment by jlprat** [comment]
Feedback applied @soarez

**Review by soarez — APPROVED** [review]

**Comment by jlprat** [comment]
No need to run the CI for this one as it won't have any impact on CI

**Pull request merged**